econml/dowhy.py

```diff
@@ -0,0 +1,222 @@
+        init_params = {}
+        for p in self._get_params():
+            init_params[p] = getattr(self._cate_estimator, p)
+        self.estimate_ = self.dowhy_.estimate_effect(self.identified_estimand_,
+                                                     method_name=method_name,
+                                                     control_value=control_value,
+                                                     treatment_value=treatment_value,
+                                                     target_units=target_units,
+                                                     method_params={
+                                                         "init_params": init_params,
+                                                         "fit_params": kwargs,
+                                                     },
+                                                     )
+        return self
+
+    def refute_estimate(self, *, method_name, **kwargs):
+        """Refute an estimated causal effect.
+
+        If method_name is provided, uses the provided method. In the future, we may support automatic
+        selection of suitable refutation tests.
+        Following refutation methods are supported:
+            * Adding a randomly-generated confounder: "random_common_cause"
+            * Adding a confounder that is associated with both treatment and outcome: "add_unobserved_common_cause"
+            * Replacing the treatment with a placebo (random) variable): "placebo_treatment_refuter"
+            * Removing a random subset of the data: "data_subset_refuter"
+
+        Parameters
+        ----------
+        method_name: string
+            Name of the refutation method
+        kwargs: optional
+            Additional arguments that are passed directly to the refutation method.
+            Can specify a random seed here to ensure reproducible results ('random_seed' parameter).
+            For method-specific parameters, consult the documentation for the specific method.
+            All refutation methods are in the causal_refuters subpackage.
+
+        Returns
+        -------
+        RefuteResult: an instance of the RefuteResult class
+        """
+        return self.dowhy_.refute_estimate(
+            self.identified_estimand_, self.estimate_, method_name=method_name, **kwargs
+        )
+
+    # We don't allow user to call refit_final from this class, since internally dowhy effect estimate will only update
+    # cate estimator but not the effect.
+    def refit_final(self, inference=None):
+        raise AttributeError(
+            "Method refit_final is not allowed here! Please call it from Cate Estimator directly! ")
+
+    def __getattr__(self, attr):
+        # don't proxy special methods
+        if attr.startswith('__'):
+            raise AttributeError(attr)
+        elif attr in ['_cate_estimator', 'dowhy_',
+                      'identified_estimand_', 'estimate_']:
+            self.__dict__[attr] = value
```

Comment:
This looks like a copy/paste of the setter code (`value` isn't even defined in this method).  If your code works, I assume that means this whole `elif` block can be removed because it's never called.

Reply:
Good catch! If remove those lines I will get an error "maximum recursion depth exceeded", and I know why now. When we search `hasattr(self.estimate_._estimator_object, attr)`, `self.estimate_` causes `__getattr__` to be called again, so it's a infinite loop.

Reply:
Hm. That's still somewhat surprising to me because `self.estimate_` should be a normal attribute, right, in which case I'd have thought that `__getattr__` would not even be called...  Or is that something that only happens before self.estimate_ is ever set for the first time?